### ISSUE TITLE
Renaming files after download

### ISSUE THREAD

**Issue by sofiedotcafe** [NickvisionApps/Parabolic#767]
Hey!

I used to use `yt-dlp`'s function to rename the music I downloaded to `Artist - Album`.
This is not possible with Parabolic?

Maybe something to just execute an `exiftool` snippet after download?
The one I currently use is:
```sh
exiftool '-FileName<${Artist;} - ${Album;}.%e'
```

**Comment by Batwam** [comment]
I also wish there was a way to run a custom shell command in the download folder at the end of the download. I personally run `beet import .` to populate lyrics and file the songs.

**Comment by nlogozzo** [comment]
Closing until more information is provided.

**Comment by kuraikaze** [comment]
That would presumably be the `-o, --output [TYPES:]TEMPLATE` argument.

I came across this issue because I too want to specify a more customisable download\output path for music, specifically ;)

The current argument I use in cmdline `yt-dlp` is:-
`--output "C:\WHATEVER\%(artist)s/%(album)s/%(playlist_index)s. %(artist)s - %(title)s.%(ext)s"`

which gives rise to an ordered directory structure of, say:-
`````
C:\WHATEVER\Famous Person\Real Album Name\1. Famous Person - First Song Name.mp3
C:\WHATEVER\Famous Person\Real Album Name\2. Famous Person - Second Song Name.mp3
C:\WHATEVER\Other Famous Person\Other Real Album Name\1. Other Famous Person - Other First Song Name.mp3
C:\WHATEVER\Other Famous Person\Other Real Album Name\2. Other Famous Person - Other Second Song Name.mp3
etc.
`````
Currently Parabolic only gives me:-
`````
C:\WHATEVER\Album - Real Album Name\1 - First Song Name.mp3
C:\WHATEVER\Album - Real Album Name\2 - Second Song Name.mp3
C:\WHATEVER\Album - Other Real Album Name\1 - Other First Song Name.mp3
C:\WHATEVER\Album - Other Real Album Name\2 - Other Second Song Name.mp3
etc.
`````
Is there any chance you might implement the `--ouput` argument in some manner, or at least allow for the addition of a way for a user to add some of the more complex `yt-dlp` arguments manually ?

**Comment by nlogozzo** [comment]
+ Bulk file renaming support for playlists

**Comment by RasoolSaeedNejad** [comment]
Hi dear friend.
In a simple program I wrote with bash based on yt dlp, for example, I used this variable
`NAME="%(upload_date)s_%(title)s_[tag%(format_id)s]_%(height)sp.%(ext)s"`

The output is as follows:
`20210910_Dimash_Stranger_tag(137+140)_1080p.mp4`

Do you have any plans to add such features to Parabolic?

**Comment by nlogozzo** [comment]
Yes that would be exactly what this would fix

**Comment by kuraikaze** [comment]
Hi Nick,
I see that you removed Advanced File Naming from the Milestone for V2025.8.0 
Does this mean the implementation of the "yt-dlp --output" option (or other similar 'in app' auto-renaming functionality) has been shelved, or am I simply being dim for not being able to find the option in 8.0/8.1 builds ?

**Comment by nlogozzo** [comment]
> Hi Nick,
> I see that you removed Advanced File Naming from the Milestone for V2025.8.0 
> Does this mean the implementation of the "yt-dlp --output" option (or other similar 'in app' auto-renaming functionality) has been shelved, or am I simply being dim for not being able to find the option in 8.0/8.1 builds ?

Yes, the feature has not been implemented yet. It's not as simple as I thought it was going to be to implement it. There is some more planning I need to go through to figure out how I will implement it.

**Comment by kuraikaze** [comment]
OK - I'll keep using the extra step of PowerToys PowerRename regex/bulk-rename tool for a while longer, and keep a weather eye out on the Parabolic commits/changelogs. :relaxed:

**Comment by influential-eliot** [comment]
Purely popping up to add inconsequential 'me too' to the feature request. :)

For me, I was purely thinking that by being able to add (for example) the following flags to my downloads I can (where possible) download different versions of the same video, and they will automatically have a different name based on ...

- `[Resolution]` - places `480p` / `1080p` / `2160p` / etc in the filename
- `[CodecV]` - places `H264` / `H265` / `HEVC` / `AV1` / etc in the filename
- `[CodecA]` - places `MP3` / `OGG` / `FLAC` / etc in the filename

Anyway, yes ... sorry, ignore me 😅
